libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java

```diff
@@ -294,10 +311,17 @@ protected void registerSoupUsingTableName(String soupName, IndexSpec[] indexSpec
 
         int i = 0;
         for (IndexSpec indexSpec : indexSpecs) {
-            // for create table
+            // Column name or expression the db index is on
             String columnName = soupTableName + "_" + i;
-            String columnType = indexSpec.type.getColumnType();
-            createTableStmt.append(", ").append(columnName).append(" ").append(columnType);
+            if (TypeGroup.value_indexed_with_json_extract.isMember(indexSpec.type)) {
+                columnName = "json_extract(" + SOUP_COL + ", '$." + indexSpec.path + "')";
```

Comment:
That's the key of it all, instead of the column name being an actual column, it's an expression.
So when creating the index / selecting on / selecting by that field, we want to use the expression instead of a column name.